examples/prometheus/prometheus.yaml

```diff
@@ -291,3 +301,196 @@ objects:
         - source_labels: [__meta_kubernetes_service_name]
           action: replace
           target_label: kubernetes_name
+
+      alerting:
+        alertmanagers:
+        - scheme: http
+          static_configs:
+          - targets:
+            - "alertmanager:9093"
+
+# Create a fully end-to-end TLS connection to the proxy
+- apiVersion: route.openshift.io/v1
+  kind: Route
+  metadata:
+    name: alerts
+    namespace: "${NAMESPACE}"
+  spec:
+    to:
+      name: alerts
+    tls:
+      termination: Reencrypt
+
+# Temporary route to alertmanager
+- apiVersion: v1
+  kind: Route
+  metadata:
+    name: alertmanager
+    namespace: "${NAMESPACE}"
+  spec:
+    port:
+      targetPort: alertmanager
+    to:
+      kind: Service
+      name: alertmanager
+      weight: 100
+    wildcardPolicy: None
+
+# Create a service to access alerts vi auth-proxy
+- apiVersion: v1
+  kind: Service
+  metadata:
+    annotations:
+      service.alpha.openshift.io/serving-cert-secret-name: alerts-tls
+    labels:
+      name: alerts
+    name: alerts
+    namespace: "${NAMESPACE}"
+  spec:
+    ports:
+    - name: alerts
+      port: 443
+      protocol: TCP
+      targetPort: 8443
+    selector:
+      app: alerts
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: alerts-proxy
+    namespace: "${NAMESPACE}"
+  stringData:
+    session_secret: "${SESSION_SECRET}="
+
+# Create an internal direct service to alertmanager app
+- apiVersion: v1
```

Comment:
To my understanding, we interact with alertmanager/prometheus via oath_proxy. We need an oauth_proxy for each app, and alertmanager resides behind the message-buffer for persistency.
Prometheus needs to send alerts to alertmanager, so this is why we expose the service.
Are you concerned with the service allowing http (not https)?
Can we use the same oauth-proxy to route 2 different apps in the same pod?